streams/test-utils/src/main/java/org/apache/kafka/streams/test/ConsumerRecordFactory.java

```diff
@@ -215,87 +254,183 @@ public void advanceTimeMs(final long advanceMs) {
             throw new IllegalStateException("ConsumerRecordFactory was created without defaultTopicName. " +
                 "Use #create(String topicName, K key, V value) instead.");
         }
-        return create(topicName, key, value);
+        return create(topicName, key, value, headers);
     }
 
     /**
      * Create a {@link ConsumerRecord} with {@code null}-key and the given topic name, value, and timestamp.
+     * Does not auto advance internally tracked time.
+     *
+     * @param topicName the topic name
+     * @param value the record value
+     * @param timestampMs the record timestamp
+     * @return the generated {@link ConsumerRecord}
+     */
+    public ConsumerRecord<byte[], byte[]> create(final String topicName,
+                                                 final V value,
+                                                 final long timestampMs) {
+        return create(topicName, null, value, new RecordHeaders(), timestampMs);
+    }
+
+    /**
+     * Create a {@link ConsumerRecord} with {@code null}-key and the given topic name, value, headers, and timestamp.
+     * Does not auto advance internally tracked time.
      *
      * @param topicName the topic name
      * @param value the record value
+     * @param headers the record headers
      * @param timestampMs the record timestamp
      * @return the generated {@link ConsumerRecord}
      */
     public ConsumerRecord<byte[], byte[]> create(final String topicName,
                                                  final V value,
+                                                 final Headers headers,
                                                  final long timestampMs) {
-        return create(topicName, null, value, timestampMs);
+        return create(topicName, null, value, headers, timestampMs);
     }
 
     /**
      * Create a {@link ConsumerRecord} with default topic name and {@code null}-key as well as given value and timestamp.
+     * Does not auto advance internally tracked time.
      *
      * @param value the record value
      * @param timestampMs the record timestamp
      * @return the generated {@link ConsumerRecord}
      */
     public ConsumerRecord<byte[], byte[]> create(final V value,
                                                  final long timestampMs) {
+        return create(value, timestampMs, new RecordHeaders());
+    }
+
+    /**
+     * Create a {@link ConsumerRecord} with default topic name and {@code null}-key as well as given value, headers and timestamp.
+     * Does not auto advance internally tracked time.
+     *
+     * @param value the record value
+     * @param headers the record headers
+     * @param timestampMs the record timestamp
+     * @return the generated {@link ConsumerRecord}
+     */
+    public ConsumerRecord<byte[], byte[]> create(final V value,
+                                                 final long timestampMs,
+                                                 final Headers headers) {
```

Comment:
the other methods take `headers` before `timestamp` -- we should be consistent.

Reply:
ack.